src/sentry/spans/consumers/process/factory.py

```diff
@@ -0,0 +1,43 @@
+from typing import Mapping
+
+import msgpack
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.commit import CommitOffsets
+from arroyo.processing.strategies.produce import Produce
+from arroyo.processing.strategies.run_task import RunTask
+from arroyo.types import Commit, Message, Partition, Topic
+from confluent_kafka import Producer
+
+from sentry.spans.span import Span
+from sentry.utils import kafka_config
+
+
+def process_message(message: Message[KafkaPayload]) -> None:
+    payload = msgpack.unpackb(message.payload.value)
+    payload["span"]["project_id"] = payload["project_id"]
+    Span.from_dict(payload).save()
+
+
+class ProcessSpansStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    def __init__(self, output_topic: str):
+        super().__init__()
+        snuba_spans = kafka_config.get_topic_definition(output_topic)
+        self.__topic = Topic(name=output_topic)
+        self.__producer = Producer(
+            kafka_config.get_kafka_producer_cluster_options(snuba_spans["cluster"]),
+        )
+
+    def create_with_partitions(
+        self,
+        commit: Commit,
+        partitions: Mapping[Partition, int],
+    ) -> ProcessingStrategy[KafkaPayload]:
+        return RunTask(
```

Comment:
`sentry.utils.arroyo.RunTaskWithMultiprocessing` is a simplified wrapper that sets up the Django app and metrics for you. I don't have a strong opinion on whether the first iteration of this consumer needs multiprocessing but I am sure single-threadedness won't be sustainable for long.

Reply:
@untitaker will the unified consumer take care of all the additional arguments I need to run the strategy (`num_processes`, `max_batch_size`, etc)?

Reply:
no, sorry. it would be cool to inject them into `RunTaskWithMultiprocessing` but for now there's only a helper function to reduce the boilerplate of defining those options (see metrics indexer e.g.)

plan was to bundle all the kwargs into a dataclass and make the user forward only that argument. another, crazier idea is to inject them via threadlocals